addons/website_sale/controllers/delivery.py

```diff
@@ -187,6 +187,55 @@ def _get_rate(carrier, order, is_express_checkout_flow=False):
                     rate['price'] = taxes['total_included']
         return rate
 
+    def order_lines_2_google_api(self, order_lines):
+        """ Transforms a list of order lines into a dict for google analytics """
+        order_lines_not_delivery = order_lines.filtered(lambda line: not line.is_delivery)
+        return super(WebsiteSaleDelivery, self).order_lines_2_google_api(order_lines_not_delivery)
+
+    def order_2_return_dict(self, order):
+        """ Returns the tracking_cart dict of the order for Google analytics """
+        ret = super(WebsiteSaleDelivery, self).order_2_return_dict(order)
+        delivery_line = order.order_line.filtered('is_delivery')
+        if delivery_line:
+            ret['shipping'] = delivery_line.price_unit
+        return ret
+
+    def _get_express_shop_payment_values(self, order, **kwargs):
+        values = super(WebsiteSaleDelivery, self)._get_express_shop_payment_values(order, **kwargs)
+        values['shipping_address_update_route'] = self._express_checkout_shipping_route
+        return values
+
+    def _get_shop_payment_values(self, order, **kwargs):
+        values = super(WebsiteSaleDelivery, self)._get_shop_payment_values(order, **kwargs)
+        has_storable_products = any(line.product_id.type in ['consu', 'product'] for line in order.order_line)
+
+        if has_storable_products:
+            if order.carrier_id and not order.delivery_rating_success:
+                order._remove_delivery_line()
+
+            delivery_carriers = order._get_delivery_methods().sudo()
+            accepted_carriers = []
+            for carrier in delivery_carriers:
+                if carrier.delivery_type == 'base_on_rule':
+                    try:
+                        carrier._get_price_available(order)
+                        accepted_carriers.append(carrier)
+                    except UserError:
+                        continue
+                else:
+                    accepted_carriers.append(carrier)
```

Comment:
This doesn't cover the express checkout flow. Delivery carriers are also listed in `express_checkout_process_shipping_address`. Please find a way to uniformize this or ensure that both controllers' behavior is the same :smiley: